File: FacturXDotNet/ApplicableHeaderTradeAgreement.cs
namespace FacturXDotNet;

/// <summary>
///     <b>HEADER TRADE AGREEMENT</b>
/// </summary>
/// <ID>BT-10-00</ID>
/// <CiiXPath>/rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeAgreement</CiiXPath>
/// <Profile>MINIMUM</Profile>
public class ApplicableHeaderTradeAgreement
{
    /// <summary>
    ///     <b>Buyer reference</b> - An identifier assigned by the Buyer used for internal routing purposes.
    /// </summary>
    /// <remarks>
    ///     The identifier is defined by the Buyer (e.g. contact ID, department, office id, project code), but provided by the Seller in the Invoice.
    /// </remarks>
    /// <ID>BT-10</ID>
    /// <CiiXPath>/rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeAgreement/ram:BuyerReference</CiiXPath>
    /// <Profile>MINIMUM</Profile>
    /// <ChorusPro>For the public sector, it is the "Service Exécutant". It is mandatory for some buyers. It must belong to the Chorus Pro repository. It is limited to 100 characters.</ChorusPro>
    public string? BuyerReference { get; set; }

    /// <inheritdoc cref="FacturXDotNet.SellerTradeParty" />
    public required SellerTradeParty SellerTradeParty { get; set; }

    /// <inheritdoc cref="FacturXDotNet.BuyerTradeParty" />
    public required BuyerTradeParty BuyerTradeParty { get; set; }

    /// <inheritdoc cref="FacturXDotNet.BuyerOrderReferencedDocument" />
    public BuyerOrderReferencedDocument? BuyerOrderReferencedDocument { get; set; }
}

File: src/FacturXDotNet.WebEditor/src/app/core/facturx-models/cii/applicable-header-trade-agreement.ts
import { SellerTradeParty } from './seller-trade-party';
import { BuyerTradeParty } from './buyer-trade-party';
import { BuyerOrderReferencedDocument } from './buyer-order-referenced-document';

/**
 * **HEADER TRADE AGREEMENT**
 *
 * @ID BT-10-00
 * @CiiXmlPath /rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeAgreement
 * @Profile MINIMUM
 */
export interface ApplicableHeaderTradeAgreement {
  /**
   * **Buyer reference** - An identifier assigned by the Buyer used for internal routing purposes.
   *
   * @remarks The identifier is defined by the Buyer (e.g. contact ID, department, office id, project code), but provided by the Seller in the Invoice.
   *
   * @ID BT-10
   * @CiiXmlPath /rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeAgreement/ram:BuyerReference
   * @Profile MINIMUM
   * @ChorusPro For the public sector, it is the "Service Exécutant". It is mandatory for some buyers. It must belong to the Chorus Pro repository. It is limited to 100 characters.
   */
  buyerReference?: string;

  /**
   * **SELLER** - A group of business terms providing information about the Seller.
   *
   * @ID BG-4
   * @CiiXmlPath /rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeAgreement/ram:SellerTradeParty
   * @Profile MINIMUM
   */
  sellerTradeParty?: SellerTradeParty;

  /**
   * **BUYER** - A group of business terms providing information about the Buyer.
   *
   * @ID BG-7
   * @CiiXmlPath /rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeAgreement/ram:BuyerTradeParty
   * @Profile MINIMUM
   */
  buyerTradeParty?: BuyerTradeParty;

  /**
   * **PURCHASE ORDER REFERENCE**
   *
   * @ID BT-13-00
   * @CiiXmlPath /rsm:CrossIndustryInvoice/rsm:SupplyChainTradeTransaction/ram:ApplicableHeaderTradeAgreement/ram:BuyerOrderReferencedDocument
   * @Profile MINIMUM
   */
  buyerOrderReferencedDocument?: BuyerOrderReferencedDocument;
}
